File: tests/rest-assured-java/src/test/java/com/portfolio/qa/payloads/TaskPayload.java
package com.portfolio.qa.payloads;

import java.util.Map;

public record TaskPayload(
    String title,
    String description,
    String status
) {

    public Map<String, Object> toMap() {
        return Map.of(
            "title", title,
            "description", description,
            "status", status
        );
    }

    public static TaskPayload validTodoTask() {
        return new TaskPayload(
            "Automatizar con Rest Assured",
            "Caso feliz CRUD desde Java.",
            "TODO"
        );
    }

    public static TaskPayload updatedInProgressTask() {
        return new TaskPayload(
            "Automatizar con Java actualizado",
            "Actualizado por PUT.",
            "IN_PROGRESS"
        );
    }

    public static TaskPayload invalidStatusTask() {
        return new TaskPayload(
            "Estado invalido",
            "Validar regla de negocio.",
            "BLOCKED"
        );
    }
}

File: tests/playwright-ts/src/payloads/task-payload.ts
export type TaskStatus = 'TODO' | 'IN_PROGRESS' | 'DONE' | 'BLOCKED';

export interface TaskPayload {
  title: string;
  description: string;
  status: TaskStatus;
}

export const TaskPayloads = {
  validTodoTask(): TaskPayload {
    return {
      title: 'Automatizar con Playwright',
      description: 'Caso feliz CRUD desde TypeScript.',
      status: 'TODO'
    };
  },

  updatedInProgressTask(): TaskPayload {
    return {
      title: 'Automatizar con Playwright actualizado',
      description: 'Actualizado por PUT desde TypeScript.',
      status: 'IN_PROGRESS'
    };
  }
};